### PR TITLE
Add utility to generate masks.

### PR DESCRIPTION
**Pull request by cornmander**
Created a new small script to generate mask files using the detectron2 library. Untested so far through the whole pipeline.

### CODE REVIEW

File: util/gen_mask.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    predictor = DefaultPredictor(cfg)
+    input_frames = dataset_path.glob('*.jpg')
+    if input_frames:
+        dest_path.mkdir(parents=True, exist_ok=True)
+    for input_frame in input_frames:
+        im = detection_utils.read_image(str(input_frame))
+        outputs = predictor(im)
+        instances = outputs['instances']
+        # get the top X predicted object masks
+        masks = outputs['instances'].pred_masks[:MAX_DETECT]
```
Comment (vidit98) [inline util/gen_mask.py:43]: Along with selecting max objects we also put a hard threshold that confidence score should be greater than 0.1.

**Review by vidit98 — COMMENTED** [review]
Please see the comment on line 43.

**Comment by akashpalrecha** [comment]
I used this code to get the predictor. My model differs from yours slightly (`1x` vs `3x`):
```python
cfg = get_cfg()
cfg.merge_from_file("../detectron2/configs/COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_1x.yaml")
cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.1
cfg.MODEL.RPN.NMS_THRESH = 0.5
cfg.MODEL.ROI_HEADS.NMS_THRESH_TEST = 0.2
cfg.MODEL.WEIGHTS = "detectron2://COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_1x/137260431/model_final_a54504.pkl"
```

Should I perhaps be using the `3x` version of the model @vidit98 ?

**Comment by shubhika03** [comment]
> I used this code to get the predictor. My model differs from yours slightly (`1x` vs `3x`):
> 
> ```python
> cfg = get_cfg()
> cfg.merge_from_file("../detectron2/configs/COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_1x.yaml")
> cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.1
> cfg.MODEL.RPN.NMS_THRESH = 0.5
> cfg.MODEL.ROI_HEADS.NMS_THRESH_TEST = 0.2
> cfg.MODEL.WEIGHTS = "detectron2://COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_1x/137260431/model_final_a54504.pkl"
> ```
> 
> Should I perhaps be using the `3x` version of the model @vidit98 ?

If you are training the Selector Net, then you can use either version of the model.  However, if you don't want to train the Selector Net, then I think the 1x version of the model is fine.

**Pull request closed without merging**